timemachine/fe/utils.py

```diff
@@ -233,7 +233,7 @@ def get_connected_components(nodes, relative_inds, absolute_inds) -> List[np.arr
         return filtered_components
 
 
-def validate_map(n_nodes: int, relative_inds: np.array, absolute_inds: np.array) -> bool:
+def validate_map(n_nodes: int, relative_inds: List[Tuple[int, int]], absolute_inds: List[int]) -> bool:
```

Comment:
Tangent / curiosity: It's often convenient to use array containers and python containers interchangeably in this sort of code.

Is there a more generic type for this situation? (should accept: int array of shape (n_pairs, 2), a list of 2-tuples, etc... Initial thought is something like `Union[Array, List[Tuple[int, int]]]`, but that seems counterproductive since more verbose, and since array typing is less precise)

Reply:
Nice: `Iterable[Tuple[int, int]]` seems both generic and precise

>(I think this sort of generalization might be best done as an iterative process; trying to come up with the most general types initially is a fair amount of work and error-prone.

Agreed!